File: code/dart/modular_api/lib/dto.dart
/// Web-safe DTO contract for modular_api.
///
/// Runtime-free surface for **defining and validating** the request/response
/// DTOs of a use case, without pulling in the HTTP server runtime. Import this
/// instead of the full `package:modular_api/modular_api.dart` barrel from
/// packages that are shared with web/desktop clients (e.g. a Flutter app's
/// `dto` package), so the shared types never drag `dart:io` into a web build.
library;

export 'src/core/usecase/usecase.dart' show Input, Output;
export 'src/core/usecase/use_case_exception.dart' show UseCaseException;
export 'src/core/schema/field.dart'
    show SchemaField, buildSchema, validateJsonFields, InputValidationException;

File: code/ts/modular_api/src/dto.ts
// dto.ts — Web-safe DTO contract entry-point.
//
// Runtime-free surface for defining and validating request/response DTOs,
// without importing the server runtime. Import via '@macss/modular-api/dto'
// from packages shared with browser/front-end code, so the shared types never
// drag Node-only globals into a web bundle. (The barrel '@macss/modular-api'
// re-exports the full framework, including the Express server, and is
// server-only.)

export { Input, Output } from './core/usecase';
export { Field, getFieldMetadata } from './core/schema/field';
export type { FieldMeta, FieldOptions } from './core/schema/field';
export { UseCaseException } from './core/use_case_exception';
export { InputValidationError } from './core/input_validation_error';
